pelican/settings.py

```diff
@@ -212,6 +222,21 @@ def get_settings_from_file(path, default_settings=DEFAULT_CONFIG):
     return get_settings_from_module(module, default_settings=default_settings)
 
 
+def get_jinja_environment(settings):
+    """Sets the environment for Jinja"""
+
+    jinja_env = settings['JINJA_ENVIRONMENT']
```

Comment:
This is unnecessary and the cause of test failures. `.setdefault` will return the value if it exists, else sets the value and returns it.